pkg/kp/kv.go

```diff
@@ -294,6 +295,11 @@ func (c consulStore) WatchPods(keyPrefix string, quitChan <-chan struct{}, errCh
 					return
 				case errChan <- util.Errorf("Could not parse pod manifest at %s: %s. Content follows: \n%s", pair.Key, err, pair.Value):
 				}
+				// do not send the partially built list, since we know it's incomplete
+				// and some callers may care about elements being removed
+				// TODO(sjung): those callers should probably be refactored to
+				// use NewKeyWatcher instead
+				continue WATCH_LOOP
```

Comment:
That's exactly correct, one bad manifest will lock down the host until it's excised. But I am not convinced it's worse:

- the bad manifest was probably put there by a deploy which means someone is expecting something to happen, so they will investigate when it goes wrong. Yes I can think of cases where a person is not going to pay attention, but the worst case scenario there is nothing happens.
- the alternative is _deleting the pod from the host_ which takes down the service itself. I would much rather have a deploy outage than an outage for the service being deployed. With the former, developers can't push new code; with the latter, you get customer visible impact.

The long term solution is to rewrite the loop altogether with `NewKeyWatcher`/`WatchNewKeys` so that the manifests are parsed after being handed out to child goroutines, but that's going to take more time to get right.